File: clojure/src/aoc_2019/day1/step1.clj
(ns aoc-2019.day1.step1
  (:gen-class))

(defn fetchList
  [fileName]
  (map read-string (clojure.string/split-lines (slurp fileName)))
)

(defn calcFuel
  [mass]
  (- (Math/floor (/ mass 3))
     2)
)

(defn calcTotalFuel
  [modules]
  (reduce + (map calcFuel modules))
)

File: nodejs/day1/step1.js
const _ = require('lodash');
const fp = require('lodash/fp');
const fs = require('fs');

// DAY 1 : ELF LAUNCH
// module launch : calcFuel(mass) => round(mass / 3) - 2
// totalFuel(modules) => sum ( map ( calcFuel modules) )
const calcFuel = (mass) => Math.floor(mass / 3) - 2;
const calcTotalFuel = (modules) => fp.reduce((acc, n) => acc + n, 0, fp.map(calcFuel, modules));

var contents = fs.readFileSync(`${__dirname}/input.txt`, 'utf8').split('\n');
console.log(calcTotalFuel(contents));